include/aws/auth/credentials.h

```diff
@@ -90,13 +92,17 @@ struct aws_credentials_provider_chain_options {
 
 struct aws_credentials_provider_imds_options {
     struct aws_client_bootstrap *bootstrap;
+    aws_credentials_provider_bootstrap_released_fn *bootstrap_release_callback;
+    void *bootstrap_release_user_data;
```

Comment:
Why is this the "bootstrap_release_callback"? Why not just have it be the "shutdown_complete_callback" and offer it on every single kind of credentials provider?

Reply:
This is intended for CRTs to signal when they are no longer using resource objects.  The other providers don't use any resources so adding callbacks (and config) seems unnecessary since they wouldn't get used.

I started with shutdown_complete and that might have been the better choice.  There's no explicit shutdown on credentials providers though.  I wonder if resources_released might be clearer.

Reply:
I would use something more generic than "bootstrap_released"
"resources_released" is fine